code/datums/components/overlay_lighting.dm

```diff
@@ -213,9 +213,13 @@
 		if(old_parent_attached_to == current_holder)
 			RegisterSignal(old_parent_attached_to, COMSIG_PARENT_QDELETING, .proc/on_holder_qdel)
 			RegisterSignal(old_parent_attached_to, COMSIG_MOVABLE_MOVED, .proc/on_holder_moved)
+			if(directional)
+				RegisterSignal(old_parent_attached_to, COMSIG_ATOM_DIR_CHANGE, .proc/on_holder_dir_change)
 	if(parent_attached_to)
 		if(parent_attached_to == current_holder)
 			UnregisterSignal(current_holder, list(COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
+			if(directional)
+				UnregisterSignal(current_holder, COMSIG_ATOM_DIR_CHANGE)
```

Comment:
This is not the right fix.
Remove every `if(directional)` check in this proc instead, this is already handled by `set_holder()`